src/main/java/hudson/plugins/svn_tag/SvnTagPlugin.java

```diff
@@ -85,7 +85,9 @@ public static boolean perform(AbstractBuild<?,?> abstractBuild,
         }
 
         SubversionSCM scm = SubversionSCM.class.cast(rootProject.getScm());
-        EnvVars envVars = rootBuild.getEnvironment(buildListener);
+        EnvVars envVars = abstractBuild.getEnvironment(buildListener);
+        EnvVars rootVars = rootBuild.getEnvironment(buildListener);
+        envVars.putAll(rootVars);
```

Comment:
This change will cause a mess in common env vars like BUILD_NUMBER.
The inclusion order is correct, so the change should not break the original behavior of the plugin.